integration-tests/src/test/java/oracle/weblogic/kubernetes/ItValidateWebhookReplicas.java

```diff
@@ -0,0 +1,288 @@
+        + "]";
+    V1Patch patch = new V1Patch(patchStr);
+    logger.info("Patching cluster resource using patch string {0} ", patchStr);
+    String responseMsg =
+        patchClusterCustomResourceReturnResponse(clusterName, domainNamespace, patch, V1Patch.PATCH_FORMAT_JSON_PATCH);
+    String expectedErrorMsg =
+        "admission webhook \"weblogic.validating.webhook\" denied the request: Change request to cluster resource '"
+            + clusterName
+            + "' cannot be honored because the replica count would exceed the cluster size '5' when patching "
+            + clusterName
+            + " in namespace "
+            + domainNamespace;
+    assertTrue(responseMsg.contains(expectedErrorMsg),
+        "Patching cluster replicas did not return the expected error msg: " + expectedErrorMsg);
+  }
+
+  /**
+   * Verify that when a domain and its clusters are running, a kubectl command that attempts to increase
+   * the domain level replicas to a value more than the max WebLogic cluster size will be rejected
+   * with a clear error message.
+   */
+  @Test
+  @DisplayName("Verify increasing the replicas of a domain too high will be rejected")
+  void testDomainReplicasTooHigh() {
+    // We need to patch the cluster resource to remove the replicas first. Otherwise, the domain level replicas will
+    // be ignored.
+    patchClusterResourceRemoveReplicas();
+
+    // patch the domain with replicas value more than max cluster size
+    String patchStr
+        = "["
+        + "{\"op\": \"replace\", \"path\": \"/spec/replicas\", \"value\": 10}"
+        + "]";
+    V1Patch patch = new V1Patch(patchStr);
+    logger.info("Patching domain resource using patch string {0} ", patchStr);
+    String response =
+        patchDomainCustomResourceReturnResponse(domainUid, domainNamespace, patch, V1Patch.PATCH_FORMAT_JSON_PATCH);
+    String expectedErrorMsg =
+        "admission webhook \"weblogic.validating.webhook\" denied the request: Change request to domain resource '"
+            + domainUid
+            + "' cannot be honored because the replica count for cluster '"
+            + clusterName
+            + "' would exceed the cluster size '5' when patching "
+            + domainUid
+            + " in namespace "
+            + domainNamespace;
+
+    assertTrue(response.contains(expectedErrorMsg),
+        String.format("Patching domain replicas did not return the expected error msg: %s, got: %s",
+            expectedErrorMsg, response));
+
+    // restore the cluster resource with replicas
+    restoreClusterResourceWithReplicas();
+  }
+
+  /**
+   * Verify that when a domain and its clusters are running, a kubectl command that attempts to change the image,
+   * as well as to increase the domain level replicas to a value that exceeds the WebLogic cluster size will NOT
+   * be rejected, but the domain will get into a Failed:true condition with a clear error message.
+   */
+  @Test
```

Comment:
Add another test methods  to verify the same usecase with (a) kubectl scale command (b) Rest API scale instead of patching the cluster Resource.